web/src/engine/websites/DayComics.ts

```diff
@@ -0,0 +1,119 @@
+import { Tags } from '../Tags';
+import icon from './DayComics.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchWindowScript } from '../platform/FetchProvider';
+
+const pageScript = `
+    new Promise ( resolve =>  resolve([...document.querySelectorAll('div#comicContent div.imgSubWrapper img')].map(image => image.dataset.src)));
+`;
+
+type APIResult<T> = {
+    data: T
+}
+
+type APIComicDetails = {
+    comic: APIComic,
+    episode : APIChapter[]
+}
+
+type APIComic = {
+    comicId: number,
+    language: string,
+    information: {
+        title: string
+    }
+}
+
+type APIComicList = {
+    new?: APIComic[],
+    binge?: APIComic[],
+    montly?: APIComic[]
+}
+
+type APIChapter = {
+    episodeId: number,
+    information: {
+        title: string,
+        subtitle : string
+    }
+}
+
+type MatureToken = {
+    token: string
+}
+
+@Common.PagesSinglePageJS(pageScript, 1500)
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    private readonly apiUrl = 'https://api.daycomics.com/api/';
+    public constructor() {
+        super('daycomics', `DayComics`, 'https://daycomics.com', Tags.Language.English, Tags.Media.Manhwa, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async Initialize(): Promise<void> {
+        //Set +18 cookie for not logged person. Logged people will have to take care of that themselves (-change account settings )
+        const { data: { token } } = await FetchJSON<APIResult<MatureToken>>(new Request('https://api.daycomics.com/preAuth/setMature', {
+            method: 'POST',
+            body: JSON.stringify({
+                mature: 1
+            }),
+            headers: {
+                'Content-Type': 'application/json',
+            }
+        }));
+
+        const cookieToken = this.EncodeString(this.FilterChars(JSON.stringify(token)), 'test');
+        return FetchWindowScript(new Request(this.URI), `window.cookieStore.set('pa_t', '${cookieToken}')`);
```

Comment:
Is it possible to call this method directly via injected JavaScript (e.g., finding the button and raise the click event handler) instead of manually fetching and encoding the cookie?